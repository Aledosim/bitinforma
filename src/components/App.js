import React from 'react'

import TopBar from '../components/TopBar'
import InfoCard from '../components/InfoCard'
import InfoCard24h from '../components/InfoCard24h'
import Background from '../components/Background'

import CurrencyProvider from '../contexts/CurrencyContext'

import styles from '../styles/components/App.module.css'

function App(pageProps) {

    return (
        <CurrencyProvider>
          <div className={styles.appContainer}>
            <div className={styles.topBarContainer}>
                <TopBar />
            </div>
            <div className={styles.backgroundContainer}>
                <Background />
            </div>
            <div className={styles.cardsContainer}>
                <InfoCard />
                <InfoCard24h />
            </div>
          </div>
        </CurrencyProvider>
    )
  }

export default App;
