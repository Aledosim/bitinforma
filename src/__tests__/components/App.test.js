import React from 'react'
import { render, waitFor } from '@testing-library/react'

import App from '../../components/App';

it('renders without crashing', async () => {
  const tree = render(<App />)

  // Check if the fetch data is rendered
  await waitFor(() => {
    expect(fetch).toHaveBeenCalledTimes(2)
  })

  expect(tree).toMatchSnapshot()
});
