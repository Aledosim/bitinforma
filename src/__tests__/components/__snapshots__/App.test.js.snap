// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="appContainer"
      >
        <div
          class="topBarContainer"
        >
          <header
            class="topBar"
            data-cy="topbar"
          >
            <div>
              <img
                alt="logo"
                class="logo"
                src="logo.svg"
              />
            </div>
            <div
              class="searchFieldContainer"
            >
              <input
                data-cy="searchField"
                id="searchField"
                placeholder="Buscar por uma moeda"
                type="text"
              />
              <img
                alt="search button"
                data-cy="searchButton"
                src="search_icon.svg"
              />
            </div>
          </header>
        </div>
        <div
          class="backgroundContainer"
        >
          <div
            class="backgroundContainer"
            data-cy="background"
          >
            <div>
              <img
                alt="background"
                src="back_up.svg"
              />
            </div>
            <div>
              <img
                alt="background"
                src="back_down.svg"
              />
            </div>
          </div>
        </div>
        <div
          class="cardsContainer"
        >
          <div
            class="infoCardContainer"
            data-cy="infocard"
          >
            <header
              class="header"
            >
              <div
                class="logoContainer"
                data-cy="coinlogo"
              >
                <span>
                  <img
                    alt="coin logo"
                    src="btc.svg"
                  />
                </span>
                <span>
                  Bitcoin
                </span>
              </div>
              <div
                class="price"
                data-cy="price"
                data-testid="price"
              >
                R$ 
                326.900
              </div>
            </header>
            <div>
              <p
                id="volume"
              >
                259,88
                 bitcoins negociados nas últimas 24hs
              </p>
              <p
                id="total"
              >
                Um total de 
                R$ 41.198.719,10
              </p>
              <div>
                <a
                  href="https://www.mercadobitcoin.com.br/negociacoes"
                >
                  Veja mais -&gt;
                </a>
              </div>
            </div>
          </div>
          <div
            class="infoCard24hContainer"
            data-cy="infocard24h"
          >
            <header
              class="header"
            >
              <div>
                <div
                  class="logoContainer"
                  data-cy="coinlogo"
                >
                  <span>
                    <img
                      alt="coin logo"
                      src="btc.svg"
                    />
                  </span>
                  <span>
                    Bitcoin
                  </span>
                </div>
              </div>
              <div>
                Nas ultimas 24h
              </div>
            </header>
            <div
              class="content"
            >
              <p>
                <span>
                  <img
                    alt="seta para cima"
                    src="up_arrow.svg"
                  />
                </span>
                <span>
                  Maior preço negociado:
                </span>
                <span>
                  380000.00000000
                </span>
              </p>
              <p>
                <span>
                  <img
                    alt="seta para baixo"
                    src="down_arrow.svg"
                  />
                </span>
                <span>
                  Menor preço negociado:
                </span>
                <span>
                  353684.14000000
                </span>
              </p>
              <p>
                <span>
                  <img
                    alt="etiqueta de preço"
                    src="price_tag.svg"
                  />
                </span>
                <span>
                  Preço da última negociação:
                </span>
                <span>
                  359999.99006000
                </span>
              </p>
              <p>
                <span>
                  <img
                    alt="preço subindo"
                    src="price_up.svg"
                  />
                </span>
                <span>
                  Menor preço de oferta de venda:
                </span>
                <span>
                  360000.00000000
                </span>
              </p>
              <p>
                <span>
                  <img
                    alt="preço descendo"
                    src="price_down.svg"
                  />
                </span>
                <span>
                  Maior preço de oferta de compra:
                </span>
                <span>
                  359999.99006000
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="appContainer"
    >
      <div
        class="topBarContainer"
      >
        <header
          class="topBar"
          data-cy="topbar"
        >
          <div>
            <img
              alt="logo"
              class="logo"
              src="logo.svg"
            />
          </div>
          <div
            class="searchFieldContainer"
          >
            <input
              data-cy="searchField"
              id="searchField"
              placeholder="Buscar por uma moeda"
              type="text"
            />
            <img
              alt="search button"
              data-cy="searchButton"
              src="search_icon.svg"
            />
          </div>
        </header>
      </div>
      <div
        class="backgroundContainer"
      >
        <div
          class="backgroundContainer"
          data-cy="background"
        >
          <div>
            <img
              alt="background"
              src="back_up.svg"
            />
          </div>
          <div>
            <img
              alt="background"
              src="back_down.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="cardsContainer"
      >
        <div
          class="infoCardContainer"
          data-cy="infocard"
        >
          <header
            class="header"
          >
            <div
              class="logoContainer"
              data-cy="coinlogo"
            >
              <span>
                <img
                  alt="coin logo"
                  src="btc.svg"
                />
              </span>
              <span>
                Bitcoin
              </span>
            </div>
            <div
              class="price"
              data-cy="price"
              data-testid="price"
            >
              R$ 
              326.900
            </div>
          </header>
          <div>
            <p
              id="volume"
            >
              259,88
               bitcoins negociados nas últimas 24hs
            </p>
            <p
              id="total"
            >
              Um total de 
              R$ 41.198.719,10
            </p>
            <div>
              <a
                href="https://www.mercadobitcoin.com.br/negociacoes"
              >
                Veja mais -&gt;
              </a>
            </div>
          </div>
        </div>
        <div
          class="infoCard24hContainer"
          data-cy="infocard24h"
        >
          <header
            class="header"
          >
            <div>
              <div
                class="logoContainer"
                data-cy="coinlogo"
              >
                <span>
                  <img
                    alt="coin logo"
                    src="btc.svg"
                  />
                </span>
                <span>
                  Bitcoin
                </span>
              </div>
            </div>
            <div>
              Nas ultimas 24h
            </div>
          </header>
          <div
            class="content"
          >
            <p>
              <span>
                <img
                  alt="seta para cima"
                  src="up_arrow.svg"
                />
              </span>
              <span>
                Maior preço negociado:
              </span>
              <span>
                380000.00000000
              </span>
            </p>
            <p>
              <span>
                <img
                  alt="seta para baixo"
                  src="down_arrow.svg"
                />
              </span>
              <span>
                Menor preço negociado:
              </span>
              <span>
                353684.14000000
              </span>
            </p>
            <p>
              <span>
                <img
                  alt="etiqueta de preço"
                  src="price_tag.svg"
                />
              </span>
              <span>
                Preço da última negociação:
              </span>
              <span>
                359999.99006000
              </span>
            </p>
            <p>
              <span>
                <img
                  alt="preço subindo"
                  src="price_up.svg"
                />
              </span>
              <span>
                Menor preço de oferta de venda:
              </span>
              <span>
                360000.00000000
              </span>
            </p>
            <p>
              <span>
                <img
                  alt="preço descendo"
                  src="price_down.svg"
                />
              </span>
              <span>
                Maior preço de oferta de compra:
              </span>
              <span>
                359999.99006000
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
